### PR TITLE
lite-xl: 2.0.1 -> 2.0.3

### PR DESCRIPTION
**Pull request by sefidel**
###### Motivation for this change
Bump versions

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/lite-xl/default.nix [hunk 1]
```diff
@@ -16,30 +16,18 @@
...
     repo = "lite-xl";
     rev = "v${version}";
-    sha256 = "sha256-+RbmT6H/5Ldhv3qOClxMjCSGMudbkGtkjo2SpGqExao=";
+    sha256 = "sha256-8Hw2zDYynWG/NwxtWgbEVUCzrtRVkOhwtsisURNqUn8=";
   };
 
   patches = [
-    # Fixes compatibility with Lua5.2, remove patch when PR merged
-    # https://github.com/lite-xl/lite-xl/pull/435
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/lite-xl/default.nix:31]: Please remove the unused fetchpatch from inputs.

Comment (boppyt) [reply pkgs/applications/editors/lite-xl/default.nix:31]: You mean the one that I replaced with the .patch file?
It was merged but not included in the v2.0.3 release, so we still need this to the package to work. Pretty sure we can remove this on v2.0.4, though.

Comment (Artturin) [reply pkgs/applications/editors/lite-xl/default.nix:31]: remove the fetchpatch in the input arguments line 3

Comment (boppyt) [reply pkgs/applications/editors/lite-xl/default.nix:31]: Ah, didn't notice that. My bad.

Comment (OPNA2608) [reply pkgs/applications/editors/lite-xl/default.nix:31]: > It was merged but not included in the v2.0.3 release, so we still need this to the package to work.

Why not keep fetching it with `fetchpatch` then? I think we generally prefer not having in-tree patches if they can be fetched from somewhere else or upstreamed.

Comment (boppyt) [reply pkgs/applications/editors/lite-xl/default.nix:31]: Patch seems to fail to automatically apply on 2.0.3, so thats why I made some changes.

**Pull request merged**